operations/configuration_management/Target_Ansible_Tasks_for_Image_Customization.md

```diff
@@ -3,29 +3,75 @@
 The Configuration Framework Service \(CFS\) enables Ansible playbooks to run against both running nodes \(node personalization\) and images prior to boot\(image customization\).
 See [Configuration Management Use Cases](Configuration_Management.md#use-cases) for more information about image customization and when it should be used.
 
-Ideally image customization playbooks should be separate from node personalization playbooks. This reduces the likelihood that Ansible content will run in both modes, and reduces the need for conditional checks in the playbooks.
+## Using `cray_cfs_image`
 
 For situations when both image customization and node personalization need to be part of the same playbook, CFS provides the `cray_cfs_image` variable to distinguish between
 node personalization and image customization. When this variable is set to `true`, it indicates that the CFS session is an image customization and the playbook is targeting an image.
 
-## Using `cray_cfs_image`
+It is best practice to include a default in Ansible roles for playbook and role portability because CFS injects this variable at runtime. This can be done in the defaults section of the role, or where the variable is called. For example:
 
-`cray_cfs_image` can be set with [Ansible playbook conditionals](https://docs.ansible.com/ansible/latest/user_guide/playbooks_conditionals.html)
-to selectively run individual tasks with `when: cray_cfs_image`, or to ignore individual tasks with `when: not cray_cfs_image`.
+```text
+{{ cray_cfs_image | default(false) }}
+```
 
-Conditionals can also be applied to entire roles if desired \(see the external [apply Ansible conditionals to roles](https://docs.ansible.com/ansible/latest/user_guide/playbooks_conditionals.html#applying-when-to-roles-imports-and-includes)\).
-In instances where the same playbook may be run in both modes, it is best practice to include a conditional on all parts of the playbook. This is best done by placing the conditional on an `include_*` statement.
-See [Write Ansible Code for CFS: Reduce wasted time](Write_Ansible_Code_for_CFS.md#reduce-wasted-time) for more information on optimizing conditionals.
+If a default is not provided, any playbooks or roles will not be runnable outside of the CFS Ansible Execution Environment \(AEE\) without the user specifying `cray_cfs_image` in the `vars` files or with the Ansible `extra-vars` options.
+
+CFS automatically sets this variable in the `hosts/01-cfs-generated.yaml` file for all sessions. When the session target is image customization, it sets `cray_cfs_image` to `true`; otherwise, it is `false`.
+
+### Using `cray_cfs_image` with `group_by`
+
+`cray_cfs_image` can be used at the top of the playbook with Ansible's `group_by` module.
+This turns the variable into a host group, and allows Ansible to more efficiently exclude tasks from being run, rather than checking a conditional for every task.
```

Comment:
Is it worth pointing out that the syntax all:&image_true is the intersection of all hosts and those hosts in the image_true group? This may be a common Ansible host pattern, but I was not familiar with it and had to look it up.

Reply:
Sure, I'll add something for this.